doc/project/release_process.rst

```diff
@@ -85,24 +85,146 @@ At that point, the whole process starts over again.
+  could be minor or moderate but, a bug in hardware that is marked as a
+  `reference hardware`` can be major or critical and in some cases a release
+  blocker.
+
+Bug priorities can be used by owners of bugs in an area or subsystem to manage
+their workflow and and priorities.
+A low or moderate bug in a platform might be set to `High Priority` by the
+owner to manage execution within a subsystem or by a hardware vendor.
+
+
+Severity
+--------
+The extent to which an issue or defect can affect the software and defines the
+impact that a given defect has on the system.
+
+The following severities shall be used:
+
+- Blocker: Such an error makes it impossible to proceed with using or testing
+  the software. See below for more details.
+- Critical: bug results in a crash of the system or one or more components of
+  the system. If the failed function is unusable and
+  there is no acceptable alternative method to achieve the advertised desired
+  outcome then the severity will be stated as critical
+- Major: As with critical, but an acceptable alternative exists
+- Moderate: Causes the system to produce incorrect, incomplete or inconsistent results
+- Minor: No crashes or damage to usability of the system and desired
+  results can be easily obtained by working around the defects
+- Cosmetic: Related to look and feel of the system
+
+Severity shall be set based on the context of the bug and not in relation to the
+overall impact on the codebase. For example, a bug in an optional subsystem that
+results in a crash shall be marked as `critical` even if the bug does not impact
+every user of Zephyr.
+
+
+Blocker Bugs
+============
+
+Blocker bug process kicks during the release process and is in effect after the
+feature freeze milestone. An issue labeled as a blocker practically blocks a
+release from happening. All blocker bugs shall be resolved before a release is
+created.
 
-The current backlog of prioritized bugs shall be used as a quality metric to
-gate the final release. The following counts shall be used:
+After hard freeze (RC3), a change to the codebase can only be merged to ‘main'
+and included in the release if it fixes a bug that is granted blocker or freeze
+exception status.
 
-.. csv-table:: Bug Count Release Thresholds
-   :header: "High", "Medium", "Low"
-   :widths: auto
+Bugs of moderate severity and higher are typically the candidates to be promoted
+to blocker bugs.
 
+The following guidelines shall be followed for release blocker bugs:
 
-   "0","<20","<50"
+- Only mark bugs as blockers if the software (Zephyr) must not be released with the bug present.
+- All collaborators can add or remove blocking labels.
+- Evaluate bugs as potential blockers based on their severity and prevalence.
+- Provide detailed rationale whenever adding or removing a blocking label.
+- Ensure all blockers have the milestone tagged.
+- Release owners have final say on blocking status; contact them with any questions.
```

Comment:
My primary problem with having High, Medium, Low and Blocker status is what is the point then of the High/Medium/Low? To me, if it is a blocker bug then it is a High bug. It has to be fixed. We used to not allow a release to occur if a High bug occurred but now we do because of this concept of a blocker. So the effect is that a Blocker is a "higher level" bug. If I would apply a numeric to this then it would be

1 - Blocker
2 - High
3 - Medium
4 - Low

Effectively 4 levels/classifications of bugs. Now I know you are really using "blocker" as an overlay modifier but I don't see the point. It is a class 1 bug that must be fixed or the release can't happen.

Reply:
I'm with David on this, not clear what we'll do with all the categorization, ultimately it's a binary decision on whether to release or not.

Reply:
I view it as an indicator to external parties to use when making an assessment on the state of the release and if they will want to upgrade to it.  The project likely will only use a binary setting, but external entities may need more detail to make a decision on an upgrade.  For example, upgrading to a release there may exist a security or safety requirement that requests we make a detailed audit of the "High" issues.  The project isn't using them, but the snapshot of state at release is something many groups use for determining upgrades.